packages/technical-features/feature-core/src/registration-of-feature.test.ts

```diff
@@ -0,0 +1,147 @@
+} from "@ogre-tools/injectable";
+import type { Feature } from "./feature";
+import { getFeature } from "./feature";
+import { deregisterFeature } from "./deregister-feature";
+
+describe("register-feature", () => {
+  describe("given di-container and a Features with injectables, and given Features are registered", () => {
+    let di: DiContainer;
+    let someInjectable: Injectable<string>;
+    let someInjectable2: Injectable<string>;
+    let someFeature: Feature;
+    let someFeature2: Feature;
+    let instance: string;
+
+    beforeEach(() => {
+      di = createContainer("irrelevant");
+
+      someInjectable = getInjectable({
+        id: "some-injectable",
+        instantiate: () => "some-instance",
+      });
+
+      someInjectable2 = getInjectable({
+        id: "some-injectable-2",
+        instantiate: () => "some-instance-2",
+      });
+
+      someFeature = getFeature({
+        id: "some-feature-1",
+        register: (di) => di.register(someInjectable),
+      });
+
+      someFeature2 = getFeature({
+        id: "some-feature-2",
+        register: (di) => di.register(someInjectable2),
+      });
+
+      registerFeature(di, someFeature);
+      registerFeature(di, someFeature2);
+    });
+
+    it("when an injectable is injected, does so", () => {
+      instance = di.inject(someInjectable);
+
+      expect(instance).toBe("some-instance");
+    });
+
+    describe("given a Feature is deregistered", () => {
+      beforeEach(() => {
+        deregisterFeature(di, someFeature);
+      });
+
+      it("when injecting a related injectable, throws", () => {
+        expect(() => {
+          di.inject(someInjectable);
+        }).toThrow();
+      });
+
+      it("when injecting an unrelated injectable, does so", () => {
+        const instance = di.inject(someInjectable2);
+
+        expect(instance).toBe("some-instance-2");
+      });
+
+      describe("given the Feature is registered again", () => {
+        beforeEach(() => {
+          registerFeature(di, someFeature);
+        });
+
+        it("when injecting a related injectable, does so", () => {
+          const instance = di.inject(someInjectable);
+
+          expect(instance).toBe("some-instance");
+        });
+
+        it("when injecting an unrelated injectable, does so", () => {
+          const instance = di.inject(someInjectable2);
+
+          expect(instance).toBe("some-instance-2");
+        });
+      });
+    });
+
+    it("when a Feature is registered again, throws", () => {
+      expect(() => {
+        registerFeature(di, someFeature);
+      }).toThrow(
+        'Tried to register feature "some-feature-1", but it was already registered.'
+      );
+    });
+
+    it("given a Feature deregistered, when deregistered again, throws", () => {
+      deregisterFeature(di, someFeature);
+
+      expect(() => {
+        deregisterFeature(di, someFeature);
+      }).toThrow(
+        'Tried to deregister feature "some-feature-1", but it was not registered.'
```

Comment:
Is there other more reliable ways to check this? I guess if i update the error message from in code the test will be broken, right?

Reply:
Yes, this will fail if you update the message. That's kinda important because we want to make sure that it fails just for this reason only.

Reply:
Yes, I mean could we detach the text from the reason? So changing text should not break the test, but still we should be able to verify the case (e.g. via `id` or some `CustomError`). Wdyt?

Reply:
I don't think it's needed here. It serves as easy way to find the meaningful unit test with quick search.